src/types/mod.rs

```diff
@@ -232,9 +288,57 @@ impl Decodable for Chat {
 
 impl Encodable for Chat {
     fn encode<E: Encoder>(&self, e: &mut E) -> Result<(), E::Error> {
+
         match self {
-            &Chat::User(ref u) => u.encode(e),
-            &Chat::Group(ref g) => g.encode(e),
+            &Chat::Private { ref id, ref first_name, ref last_name, ref username } => {
+                e.emit_struct("Chat", 5, |e| {
+                    try!(e.emit_struct_field("id", 0, |e| {
+                        id.encode(e)
+                    }));
+                    try!(e.emit_struct_field("type", 1, |e| {
+                        "private".encode(e)
+                    }));
+                    try!(e.emit_struct_field("first_name", 2, |e| {
+                        first_name.encode(e)
+                    }));
+                    try!(e.emit_struct_field("last_name", 3, |e| {
+                        last_name.encode(e)
+                    }));
+                    try!(e.emit_struct_field("type", 4, |e| {
+                        username.encode(e)
+                    }));
+                    Ok(())
+                })
+            },
+            &Chat::Group { ref id, ref title, ref is_supergroup} => {
```

Comment:
You could drop the `ref` on `is_supergroup` and `id`, since those are copy types. Also applicable to `id` in the other two match arms. Then you have to remove the `*` in front of those variables of course. Maybe a bit more readable...